File: mobile/lib/data/services/engine_start_mode.dart
import 'package:shared_preferences/shared_preferences.dart';

/// How the engine-start control fires: hold-to-confirm or single press.
enum EngineStartMode {
  holdConfirm('hold_confirm', 'Hold to confirm'),
  singlePress('single_press', 'Single press');

  const EngineStartMode(this.wireValue, this.label);
  final String wireValue;
  final String label;

  static const String prefsKey = 'wheeldeck.engine_start_mode';
  static const EngineStartMode fallback = holdConfirm;

  static EngineStartMode fromWireValue(String? v) =>
      EngineStartMode.values.firstWhere(
        (m) => m.wireValue == v,
        orElse: () => fallback,
      );

  static Future<EngineStartMode> load() async {
    final prefs = await SharedPreferences.getInstance();
    return fromWireValue(prefs.getString(prefsKey));
  }

  Future<void> save() async {
    final prefs = await SharedPreferences.getInstance();
    await prefs.setString(prefsKey, wireValue);
  }
}

File: android/app/src/main/kotlin/dev/fazrigading/wheeldeck/data/services/EngineStartMode.kt
package dev.fazrigading.wheeldeck.data.services

/// How the engine-start control fires: hold-to-confirm or single press.
enum class EngineStartMode(val wireValue: String, val label: String) {
    HoldConfirm("hold_confirm", "Hold to confirm"),
    SinglePress("single_press", "Single press");

    companion object {
        const val KEY = "wheeldeck.engine_start_mode"
        val fallback = HoldConfirm

        fun fromWireValue(value: String?): EngineStartMode =
            entries.firstOrNull { it.wireValue == value } ?: fallback
    }
}
